platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/user/configuration/WorkflowUserConfigurationValidator.java

```diff
@@ -0,0 +1,67 @@
+package com.softicar.platform.workflow.module.workflow.user.configuration;
+
+import com.softicar.platform.emf.validation.AbstractEmfValidator;
+import com.softicar.platform.workflow.module.WorkflowI18n;
+import java.util.List;
+
+public class WorkflowUserConfigurationValidator extends AbstractEmfValidator<AGWorkflowUserConfiguration> {
+
+	@Override
+	protected void validate() {
+
+		assertSourceAndTargetUserNotEqual();
+		validateThisUserIsNotASubstitute();
+		if (tableRow.getSubstituteFrom() != null && tableRow.getSubstituteTo() != null) {
+			verifyValidFromBeforeValidTo();
+		}
+		if (tableRow.getSubstitute() == null) {
```

Comment:
I think this condition should be integrated into `verifyEmptyValidFromAndValidTo`